### ISSUE TITLE
populate-user.js do not get settings

### ISSUE THREAD

**Issue by kulakowka** [feathersjs/feathers-authentication#86]
This hook do not get settings for `local auth` from my `config/default.json`.

I have added configuration to default.json file.
```json
{
  "host": "localhost",
  "port": 3001,
  "postgres": "postgres://postgres:@localhost:5432/feathers",
  "auth": {
    "token": {
      "secret": "bGGBbrJRXyIPRQW3BLaVp/P9/ounEv2L7qIVGfw3uVePt1eUg4Fi6Ji6Bpjknm9rh9hKHaMLwW7aGBVCejXIxA=="
    },
    "local": {
      "usernameField": "email",
      "userEndpoint": "/api/v1/users",
      "idField": "username"
    }
  }
}
```

And i passed settings to feather-authentication:
```javascript
const authentication = require('feathers-authentication')

module.exports = function () {
  const app = this

  let config = app.get('auth')

  app.set('auth', config)
  app.configure(authentication(config))
}
```

**Comment by kulakowka** [comment]
I also found that `options.idField` parameter is not actually used. It should be like this:

```javascript
/**
 * Populate the current user associated with the JWT
 */
const defaults = {
  userEndpoint: '/users',
  passwordField: 'password',
  idField: '_id'
};

export default function(options = {}){
  options = Object.assign({}, defaults, options);

  return function(hook) {
    // If we already have a current user just pass through
    if (hook.params.user) {
      return Promise.resolve(hook);
    }

    let id;
    
    // If it's an after hook grab the id from the result
    if (hook.type === 'after') {
      id = hook.result[options.idField];
    }
    // Check to see if we have an id from a decoded JWT
    else if (hook.params.payload) {
      id = hook.params.payload[options.idField];
    }

    // If we didn't find an id then just pass through
    if (id === undefined) {
      return Promise.resolve(hook);
    }

    return new Promise(function(resolve, reject){
      hook.app.service(options.userEndpoint).get(id, {}).then(user => {
        // attach the user to the hook for use in other hooks or services
        hook.params.user = user;

        // If it's an after hook attach the user to the response
        if (hook.result) {
          hook.result.data = Object.assign({}, user = !user.toJSON ? user : user.toJSON());
          
          // format response
          delete hook.result[options.idField];
          delete hook.result.data[options.passwordField];
        }
        
        return resolve(hook);
      }).catch(reject);
    });
  };
}
```

I'll try to do PR with the necessary changes.

**Comment by ekryski** [comment]
This was closed by #87
